src/bt-pan-io.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (C) 2015  Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <fdio/task.h>
+#include <hardware/bluetooth.h>
+#include <hardware/bt_pan.h>
+#include <stdlib.h>
+#include "compiler.h"
+#include "log.h"
+#include "bt-proto.h"
+#include "bt-pdubuf.h"
+#include "bt-core-io.h"
+#include "bt-pan-io.h"
+
+#define IFNAME_LEN 17 /* interface name */
```

Comment:
Instead of a defined constant, I would prefer

  static const size_t IFNAME_LEN = 17;

if possible.